### DISCUSSION TITLE
Cannot run python file

### DISCUSSION THREAD

**Discussion by OoWitYeeMg** [leokhoa/laragon#644] [Q&A]
I already install the python program and can run in terminal of the laragon. But after tried to run in web browser the file is downloaded. I change the www document route to code. My file is in code\MIBA\index.py. I just use simple python code to test.

<img width="960" alt="image" src="https://github.com/leokhoa/laragon/assets/111106217/3c52488e-9054-4502-8f2d-3d565d77d8f9">

<img width="958" alt="image" src="https://github.com/leokhoa/laragon/assets/111106217/9bc44c70-92d8-4747-a8d1-168be6c64175">

**Comment by Pen-y-Fan** [comment]
You will need to Configure Apache to run Python CGI. See:

- [Configuring the Apache Web Server to Run Python on Windows](https://editrocket.com/articles/python_apache_windows.html)

To open httpd.conf in Laragon Menu > Apache > httpd.conf

The above blog doesn't say anything about DirectoryIndex, you may need to edit that line too [source SO: Python index page on apache2](https://stackoverflow.com/questions/69449537/python-index-page-on-apache2):

```diff
- DirectoryIndex index.html index.php
+ DirectoryIndex index.html index.php index.py
```

I haven't tried it, I hope this helps  good luck.
